src/components/MapComponent.vue

```diff
@@ -0,0 +1,335 @@
+            // Create a query where the ID equals the given ID
+              const query = blockLayer.createQuery();
+              query.where = `ID = ${this.activeBlockId}  AND RESERVED = 1`;
+              blockLayer.queryFeatures(query)
+                .then((response) => {
+                // ID is a key so there should only be one item in the
+                // features array that is returned
+                  const blockFeatures = response.features;
+                  // Sets what the popup should look like
+                  if (blockFeatures.length > 0) {
+                    blockFeatures[0].popupTemplate = this.blockPopupTemplate;
+                  }
+                  this.view.popup.open({
+                    features: blockFeatures,
+                  });
+                });
+            }
+          });
+
+          // Hides the completed block action when a block is completed
+          this.view.popup.watch('selectedFeature', (graphic) => {
+            if (graphic) {
+              const graphicTemplate = graphic.getEffectivePopupTemplate();
+              if (graphic.attributes.RESERVED === 2) {
+                graphicTemplate.actions.items[1].visible = false;
+              } else if (graphic.attributes.RESERVED === 1) {
+                graphicTemplate.actions.items[1].visible = true;
+              }
+            }
+          });
+
+          // Determine to which list the block in the popup will be added
+          // https://developers.arcgis.com/javascript/latest/sample-code/popup-actions/index.html
+          this.view.popup.on('trigger-action', (event) => {
+          // Execute the measureThis() function if the measure-this action is clicked
+          // If the event id matches one of the ids defined as an action for selecting a block
+          // then add to the list with a corresponding id
+            if (event.action.id === 'reserve'
+            || event.action.id === 'release'
+            || event.action.id === 'complete') {
+              this.pushBlock(this.view.popup.selectedFeature.attributes.ID, event.action.id);
+            }
+          });
+        });
+    },
+  },
+
+  mounted() {
+    this.$store.dispatch('getReservedBlocks');
+    this.loadMap();
+  },
+
+  // https://developers.arcgis.com/javascript/latest/api-reference/esri-layers-FeatureLayer.html#refresh
+  beforeDestroy() {
+    if (this.view) {
+      // destroy the map view
+      this.view.container = null;
+    }
+  },
+  watch: {
+    reservedBlocks() {
+      this.loadMap();
+    },
+
+    reservedFilter() {
+      this.loadMap();
+    },
+
+    labelsVisible() {
+      this.loadMap();
+    },
+  },
+};
+
+</script>
+
+<style lang='scss' scoped>
+  button.label, button.label:hover, button.label:focus {
+    position: absolute;
+    z-index: 100;
+    right: 5px;
+    top: 5px;
+    float: right;
+    background: #9AC356;
+    color: white;
+    border: none;
+    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+  }
+</style>
```

Comment:
Is there any way we could reduce the size of this file and abstract things to other files?

Reply:
Yeah at least all of the constants can be moved. There could maybe be some more complicated component rearrangement but that might be its own ticket.